src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/SaveWithFormDialogToolbarAction.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+import React from 'react';
+import type {ElementRef} from 'react';
+import {action, observable} from 'mobx';
+import Dialog from '../../../components/Dialog';
+import {default as FormContainer, memoryFormStoreFactory, ResourceFormStore} from '../../../containers/Form';
+import type {FormStoreInterface} from '../../../containers/Form';
+import Router from '../../../services/Router';
+import {translate} from '../../../utils/Translator';
+import Form from '../Form';
+import AbstractFormToolbarAction from './AbstractFormToolbarAction';
+
+export default class SaveWithFormDialogToolbarAction extends AbstractFormToolbarAction {
+    @observable showDialog: boolean = false;
+    dialogForm: ?FormContainer;
+    dialogFormStore: FormStoreInterface;
+
+    constructor(
+        resourceFormStore: ResourceFormStore,
+        form: Form,
+        router: Router,
+        locales: ?Array<string>,
+        options: {[key: string]: mixed}
+    ) {
+        super(resourceFormStore, form, router, locales, options);
+
+        const {formKey} = options;
+
+        if (typeof formKey !== 'string') {
+            throw new Error('The "formKey" option of the SaveWithFormDialogToolbarAction must be a string!');
+        }
+
+        this.dialogFormStore = memoryFormStoreFactory.createFromFormKey(formKey);
+    }
+
+    handleConfirm = () => {
+        if (!this.dialogForm) {
+            throw new Error('The dialog form was not initialized. This should not happen and is likely a bug.');
+        }
+
+        this.dialogForm.submit();
+    };
+
+    @action handleCancel = () => {
+        this.showDialog = false;
+    };
+
+    @action handleSubmit = () => {
+        this.form.submit(this.dialogFormStore.data);
+        this.showDialog = false;
+    };
+
+    setDialogFormRef = (dialogForm: ?ElementRef<typeof FormContainer>) => {
+        this.dialogForm = dialogForm;
+    };
+
+    getNode() {
+        const {title} = this.options;
+
+        if (typeof title !== 'string') {
+            throw new Error('The "title" option of the SaveWithFormDialogToolbarAction must be a string!');
+        }
+
+        return (
+            <Dialog
+                cancelText={translate('sulu_admin.cancel')}
+                confirmText={translate('sulu_admin.ok')}
+                key="sulu_admin.save_with_form_dialog"
+                onCancel={this.handleCancel}
+                onConfirm={this.handleConfirm}
+                open={this.showDialog}
+                title={title}
+            >
+                <FormContainer
+                    onSubmit={this.handleSubmit}
+                    ref={this.setDialogFormRef}
+                    store={this.dialogFormStore}
+                />
```

Comment:
In the projects we did mostly lazy initialize the form store and loading of the data maybe that would also make sense here for performance/memory reasons:

```js

                {
                    this.showDialog ?
                <FormContainer
                    onSubmit={this.handleSubmit}
                    ref={this.setDialogFormRef}
                    store={this.getDialogFormStore()}
                /> : null
}

    getDialogFormStore() {
        if (!this.dialogFormStore) {
            this.dialogFormStore = ...
        }

        return this.dialogFormStore;
    }

    destroy() {
        if (this.dialogFormStore) {
            this.dialogFormStore.destroy();
        }
    }
```

Reply:
Tried to implement it this way, and while it might be better for performance/memory reasons, it comes with two downsides, which make me think that it's not worth the tradeoff:

- When the dialog is opened it will take some more time to load the metadata, otherwise it is already loaded and can be immediately displayed (better perceived performance)
- When the dialog is closed the form disappears before the dialog, which makes it feel a bit uneasy (could be fixed, but makes the code much more complicated by holding two different `showDialog` states

Reply:
Okay for me.